test/specs/print/PrintLayerService.spec.js

```diff
@@ -0,0 +1,404 @@
+          graphicXOffset: -2.205,
+          graphicYOffset: -2.205,
+          fillColor: '#ffffff',
+          fillOpacity: 0.4,
+          strokeWidth: 0.75,
+          strokeColor: '#3399cc',
+          strokeOpacity: 1,
+          strokeLinecap: 'round',
+          strokeLinejoin: 'round' }
+        );
+      });
+    });
+
+    describe('#encodeWMTS()', function() {
+
+      var extent = [420000, 30000, 900000, 350000];
+
+      var config = {
+        "attribution": "swisstopo",
+        "chargeable": true,
+        "searchable": false,
+        "format": "jpeg",
+        "hasLegend": true,
+        "serverLayerName": "ch.swisstopo.pixelkarte-farbe",
+        "attributionUrl": "https://www.swisstopo.admin.ch/fr/home.html",
+        "tooltip": false,
+        "parentLayerId": "ch.swisstopo.pixelkarte-farbe",
+        "highlightable": true,
+        "background": true,
+        "timestamps": ["current"],
+        "topics": "api,swissmaponline,wms-naz,wms-swisstopowms",
+        "resolutions": [1, 0.5, 0.25],
+        "label": "ch.swisstopo.pixelkarte-farbe_wmts",
+        "type": "wmts",
+        "timeEnabled": false,
+        "minResolution": 1.5
+      };
+
+      var options = {
+        origin: [420000, 350000],
+        tileSize: 256,
+        matrixIds: ["0", "1", "2", "3", "4", "5", "6", "7", "8", "9", "10", "11", "12", "13", "14", "15", "16", "17", "18", "19", "20", "21", "22", "23", "24", "25"],
+        resolutions: [4000, 3750, 3500, 3250, 3000, 2750, 2500, 2250, 2000, 1750, 1500, 1250, 1000, 750, 650, 500, 250, 100, 50, 20, 10, 5, 2.5, 2, 1.5],
+      }
+
+      var tileGrid = new ol.tilegrid.WMTS(options);
+
+      var source = new ol.source.WMTS({
+        url: '//wmts100.geo.admin.ch/1.0.0/{Layer}/default/current/21781/{TileMatrix}/{TileCol}/{TileRow}.jpeg',
+        tileGrid: tileGrid,
+        projection: "EPSG:21781",
+        layer: config.serverLayerName,
+        requestEncoding: 'REST',
+        dimensions: {
+          "Time": "current"
+        }
+      });
+
+      var layer = new ol.layer.Tile({
+        source: source,
+        extent: [420000, 30000, 900000, 350000]
+      });
+
+      it('returns an encoded WMTS layer', function() {
+        var encWMTS = gaPrintLayer.encodeWMTS(layer, config);
+
+        expect(encWMTS).to.eql({
+          "layer": "ch.swisstopo.pixelkarte-farbe",
+          "opacity": 1,
+          "type": "WMTS",
+          "version": "1.0.0",
+          "requestEncoding": "REST",
+          "formatSuffix": "jpeg",
+          "style": "default",
+          "dimensions": ["TIME"],
+          "params": {
+            "TIME": "current"
+          },
+          "matrixSet": "21781",
+          "baseURL": "https://wmts.geo.admin.ch",
+          "zoomOffset": 0,
+          "tileOrigin": [420000, 350000],
+          "tileSize": [256, 256],
+          "resolutions": [4000, 3750, 3500, 3250, 3000, 2750, 2500, 2250, 2000, 1750, 1500, 1250, 1000, 750, 650, 500, 250, 100, 50, 20, 10, 5, 2.5, 2, 1.5],
```

Comment:
you could define these variables once at the top of this file. They are repeated over and over again.